src/main/scala/model/TodoServer.scala

```diff
@@ -28,14 +28,29 @@ class TodoServer() {
 
   server.addConnectListener(new ConnectionListener(this))
   server.addEventListener("add_task", classOf[String], new AddTaskListener(this))
+  server.addEventListener("update_tasks", classOf[Nothing], new UpdateTaskListener(this))
   server.addEventListener("complete_task", classOf[String], new CompleteTaskListener(this))
+  server.addEventListener("register_user", classOf[String], new RegisterUserListener(this))
+  server.addEventListener("share", classOf[String], new ShareMessageListener(this))
 
   server.start()
 
-  def tasksJSON(): String = {
-    val tasks: List[Task] = database.getTasks
-    val tasksJSON: List[JsValue] = tasks.map((entry: Task) => entry.asJsValue())
-    Json.stringify(Json.toJson(tasksJSON))
+  def tasksJSON(user: String): String = {
+    if (user == "") {
+      val tasks: List[Task] = database.getTasks
+      val tasksJSON: List[JsValue] = tasks.map((entry: Task) => entry.asJsValue())
+      val jsonString = Json.stringify(Json.toJson(tasksJSON))
+//      println(jsonString)
+      jsonString
+    }
+    else {
+      val tasks: List[Task] = database.getTasks.filter(_.user == user)
+      val tasksJSON: List[JsValue] = tasks.map((entry: Task) => entry.asJsValue())
+      val jsonString = Json.stringify(Json.toJson(tasksJSON))
+//      println(jsonString)
+      jsonString
+    }
+
```

Comment:
once a user enters their username, all the information that they entered that corresponds to their username, if the username already exists, will be loaded